backend/question-service/src/questions/questions.service.spec.ts

```diff
@@ -1,18 +1,109 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import { QuestionsService } from './questions.service';
+import { QuestionDB } from './questions.model';
+import { CreateQuestionDto } from './dto/create-question.dto';
+import { UpdateQuestionDto } from './dto/update-question.dto';
+import { FilterQuestionDto } from './dto/filter-question.dto';
+import { QuestionCategory, QuestionComplexity } from './types/question.types';
+import mongoose from 'mongoose';
+
+jest.mock('./questions.model');
```

Comment:
Perhaps we will also need a questions.model.spec.ts to test the model, because in these 2 test files, the service and the model are being mocked respectively, so we haven't really tested if the CRUD operations are actually working correctly.